### DISCUSSION TITLE
error unable to decode disk's description as JSON

### DISCUSSION THREAD

**Discussion by artooro** [vmware-tanzu/velero#3251] [Community support Q&A] [answered]
**What steps did you take and what happened:**
Created a backup via the command `velero backup create initial-backup` after a fresh install using the `velero-plugin-for-gcp` plugin.

**What did you expect to happen:**
I expected the backup to complete without errors.

**The output of the following commands will help us better understand what's going on**:
- `velero backup describe initial-backup`
```
Name:         initial-backup
Namespace:    velero
Labels:       velero.io/storage-location=default
Annotations:  velero.io/source-cluster-k8s-gitversion=v1.17.13-gke.1400
              velero.io/source-cluster-k8s-major-version=1
              velero.io/source-cluster-k8s-minor-version=17+

Phase:  PartiallyFailed (run `velero backup logs initial-backup` for more information)

Errors:    1
Warnings:  0

Namespaces:
  Included:  *
  Excluded:  <none>

Resources:
  Included:        *
  Excluded:        <none>
  Cluster-scoped:  auto

Label selector:  <none>

Storage Location:  default

Velero-Native Snapshot PVs:  auto

TTL:  720h0m0s

Hooks:  <none>

Backup Format Version:  1.1.0

Started:    2020-11-25 12:22:26 -0500 EST
Completed:  2020-11-25 12:23:01 -0500 EST

Expiration:  2020-12-25 12:22:26 -0500 EST

Total items to be backed up:  1516
Items backed up:              1516

Velero-Native Snapshots:  17 of 17 snapshots completed successfully (specify --details for more information)
```

- `velero backup logs initial-backup | grep -v 'level=info'`
```
time="2020-11-25T17:22:33Z" level=error msg="unable to decode disk's description as JSON, so only applying Velero-assigned tags to snapshot" backup=velero/initial-backup cmd=/plugins/velero-plugin-for-gcp error="unexpected end of JSON input" logSource="/go/src/github.com/vmware-tanzu/velero-plugin-for-gcp/velero-plugin-for-gcp/volume_snapshotter.go:282" pluginName=velero-plugin-for-gcp
```

**Anything else you would like to add:**

The error message doesn't appear to be backup failure as it sounds like the disk snapshot is still being created, but the backup status says `PartiallyFailed` so it's hard to understand how critical this error is. 

**Environment:**

- Velero version (use `velero version`): v1.5.2
- Velero features (use `velero client config get features`): <NOT SET>
- Kubernetes version (use `kubectl version`): v1.17.13-gke.1400
- Kubernetes installer & version: 
- Cloud provider or hardware configuration: GKE
- OS (e.g. from `/etc/os-release`): macOS 11.0.1

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by dsu-igeek** [comment, marked as answer]
The output of the following commands will help us better understand what's going on:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`

**Reply by carlisia** [reply]
Hey @artooro are you able to give us the requested output?

**Reply by artooro** [reply]
Give me a day or so, I believe this error is related to when GCE virtual disks are encrypted using a customer supplied key.

**Reply by carlisia** [reply]
This discussion has been moved to an issue: https://github.com/vmware-tanzu/velero/issues/3348.
